tools/Azure.Mcp.Tools.ApplicationInsights/src/Commands/AppTrace/AppTraceListCommand.cs

```diff
@@ -0,0 +1,165 @@
+
+        Example usage:
+        Filter to dependency failures
+        "table": "dependencies",
+        "filters": ["success=\"false\""]
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new() { Destructive = false, Idempotent = true, LocalRequired = false, OpenWorld = false, Secret = false, ReadOnly = true };
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+        {
+            return context.Response;
+        }
+
+        AppTraceListOptions options = BindOptions(parseResult);
+        try
+        {
+            IApplicationInsightsService service = context.GetService<IApplicationInsightsService>();
+            AppListTraceResult traces = await service.ListDistributedTracesAsync(
+                subscription: options.Subscription!,
+                resourceGroup: options.ResourceGroup,
+                resourceName: options.ResourceName,
+                resourceId: options.ResourceId,
+                filters: options.Filters,
+                table: options.Table!,
+                startTime: options.StartTimeUtc!.Value,
+                endTime: options.EndTimeUtc!.Value,
+                tenant: options.Tenant,
+                options.RetryPolicy);
+
+            context.Response.Results = traces is not null ?
+                ResponseResult.Create(new AppTraceListCommandResult(traces), ApplicationInsightsJsonContext.Default.AppTraceListCommandResult) :
+                null;
+        }
+        catch (Exception ex)
+        {
+            // Log error with all relevant context
+            _logger.LogError(ex,
+                "Error in {Name}. Subscription: {Subscription}, ResourceGroup: {ResourceGroup}, ResourceName: {ResourceName}, Options: {@Options}",
+                Name, options.Subscription, options.ResourceGroup, options.ResourceName, options);
+            HandleException(context, ex);
+        }
+
+        return context.Response;
+    }
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        command.Options.Add(OptionDefinitions.Common.ResourceGroup.AsOptional());
+        command.Options.Add(ResourceName);
+        command.Options.Add(ResourceId);
+        command.Options.Add(Table);
+        command.Options.Add(Filters);
+        command.Options.Add(StartTime);
+        command.Options.Add(EndTime);
+
+        command.Validators.Add(ResourceNameOrIdRequired);
+        command.Validators.Add(TimeRangeValid);
+        command.Validators.Add(TableNameValid);
+    }
+
+    protected override AppTraceListOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.ResourceGroup ??= parseResult.GetValueOrDefault(OptionDefinitions.Common.ResourceGroup);
```

Comment:
The all need to use the .Name property to make sure it finds the right instance

Reply:
Interesting. I'll take a look.